File: challenges/3-area-trapezio/c++/baka-noob/area-trapezio.cpp
#include <iostream>

using namespace std;

int main() {

    float maior, menor, altura;
    cout << "Base Maior: ";
    cin >> maior;
    cout << "Base Menor: ";
    cin >> menor;
    cout << "Altura: ";
    cin >> altura;

    altura = ((maior + menor) * altura)/2;;

    cout << "A área do trapézio é: " << altura << endl;
    
    return 0;
}

File: challenges/3-area-trapezio/elixir/murilosrg/area_trapezio.ex
defmodule AreaTrapezio do
  def build do
    larger = read("base maior")
    smaller = read("base menor")
    height = read("altura")
    base = (larger + smaller) * height / 2

    IO.puts("A área do trapézio é: #{base}")
  end

  defp read(number) do
    IO.gets("Digite a #{number}: ")
    |> Float.parse()
    |> validate(number)
  end

  defp validate(:error, number) do
    IO.puts("Informe um numero valido")
    read(number)
  end

  defp validate({value, _}, _), do: value
end

AreaTrapezio.build()
